stablehlo/dialect/Base.cpp

```diff
@@ -300,6 +300,7 @@ FailureOr<TensorType> inferTypeWithCustomFn(
         Optional<Location>, int64_t, int64_t, int64_t, int64_t, int64_t)>
         inferDimAndBoundFn) {
   auto rank = rankedTypes[0].getRank();
+  if (rankedTypes[1].getRank() != rank) return failure();
```

Comment:
Should we check for all  i `rankedTypes[i].getRank() != rank`

Reply:
Yep.